aws-android-sdk-core/src/main/java/com/amazonaws/util/VersionInfoUtils.java

```diff
@@ -131,6 +131,9 @@ static String userAgent() {
      * @return the input with spaces replaced by underscores
      */
     private static String replaceSpaces(final String input) {
-        return input.replace(' ', '_');
+        if (input != null){
+            return input.replace(' ', '_');
+        }
+        return input;
```

Comment:
If we return null value here, `buffer.append(null)` will still throw an exception. We can return string "null" or empty string.